src/util/tab-events.js

```diff
@@ -40,18 +58,24 @@ export function whenPageLoadComplete({tabId}) {
         if (tab.status === 'complete')
             return // Resolve directly
 
-        return eventToPromise({
-            resolve: {
-                event: browser.tabs.onUpdated,
-                filter: (changedTabId, {status}) =>
-                    (changedTabId === tabId && status === 'complete'),
-            },
-            reject: {
-                event: browser.tabs.onRemoved,
-                filter: (closedTabId) => (closedTabId === tabId),
-                reason: {message: "Tab was closed before loading was complete."},
-            },
-        })
+        return new Promise((resolve, reject) => {
+            eventToPromise({
+                resolve: {
+                    event: browser.tabs.onUpdated,
+                    filter: (changedTabId, {status}) =>
+                        (changedTabId === tabId && status === 'complete'),
+                },
+                reject: {
+                    event: browser.tabs.onRemoved,
+                    filter: (closedTabId) => (closedTabId === tabId),
+                    reason: {message: "Tab was closed before loading was complete."},
+                },
+            }).then(()=>resolve())
+            .catch(()=>reject());
```

Comment:
this should probably be ```.catch(reject)``` otherwise the reject reason will be lost making quite hard to debug